flink-formats/flink-parquet/src/main/java/org/apache/flink/formats/parquet/utils/ParquetSchemaConverter.java

```diff
@@ -0,0 +1,242 @@
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.formats.parquet.utils;
+
+import org.apache.flink.api.common.typeinfo.BasicArrayTypeInfo;
+import org.apache.flink.api.common.typeinfo.BasicTypeInfo;
+import org.apache.flink.api.common.typeinfo.TypeInformation;
+import org.apache.flink.api.java.typeutils.MapTypeInfo;
+import org.apache.flink.api.java.typeutils.ObjectArrayTypeInfo;
+import org.apache.flink.api.java.typeutils.RowTypeInfo;
+
+import org.apache.parquet.avro.AvroSchemaConverter;
+import org.apache.parquet.schema.GroupType;
+import org.apache.parquet.schema.MessageType;
+import org.apache.parquet.schema.OriginalType;
+import org.apache.parquet.schema.PrimitiveType;
+import org.apache.parquet.schema.Type;
+import org.apache.parquet.schema.Types;
+
+import java.lang.reflect.Array;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Schema converter converts Parquet schema to and from Flink internal types.
+ */
+public class ParquetSchemaConverter {
+	public static final String MAP_KEY = "key";
+	public static final String MAP_VALUE = "value";
+	public static final String LIST_ELEMENT = "array";
+	public static final String MESSAGE_ROOT = "root";
+	private static final AvroSchemaConverter SCHEMA_CONVERTER = new AvroSchemaConverter();
+
+	public static TypeInformation<?> fromParquetType(MessageType type) {
+		return convertFields(type.getFields());
+	}
+
+	public static MessageType toParquetType(TypeInformation<?> typeInformation) {
+		return (MessageType) convertField(null, typeInformation, Type.Repetition.OPTIONAL);
+	}
+
+	private static TypeInformation<?> convertFields(List<Type> parquetFields) {
+		List<TypeInformation<?>> types = new ArrayList<>();
+		List<String> names = new ArrayList<>();
+		for (Type field : parquetFields) {
+			TypeInformation<?> subType = convertField(field);
+			if (subType != null) {
+				types.add(subType);
+				names.add(field.getName());
+			}
+		}
+
+		return new RowTypeInfo(types.toArray(new TypeInformation<?>[types.size()]),
+			names.toArray(new String[names.size()]));
+	}
+
+	private static TypeInformation<?> convertField(final Type fieldType) {
+		TypeInformation<?> typeInfo = null;
+		if (fieldType.isPrimitive()) {
+			PrimitiveType primitiveType = fieldType.asPrimitiveType();
```

Comment:
This function converts a parquet primitive type to a corresponding default Flink type. The explicitly type conversion to SqlTimeTypeInfo or other types probably can handled by users when there is a need. Otherwise, we need to bring user's conversion preference in the interface.

Reply:
IMHO I think we should aim to map all Parquet types to Flink types. Otherwise the integration is not end-to-end and requires a lot of customization nevertheless. New supported formats such as ORC, Avro, or JSON support all types. Which makes the connector robust.